extensions/applicationinsights-cfgsync-js/src/CfgSyncPlugin.ts

```diff
@@ -0,0 +1,419 @@
+                    }
+                }
+                return false;
+            }
+
+            function _eventOff() {
+                try {
+                    let global = getGlobal();
+                    if (global) {
+                        eventOff(global, null, null,  _evtNamespace);
+                    }
+                } catch (e) {
+                    // eslint-disable-next-line no-empty
+                }
+            }
+
+            function _sendCfgsyncEvents(customDetails?: any) {
+                try {
+                    if (!!_overrideSyncFn && isFunction(_overrideSyncFn)) {
+                        return _overrideSyncFn(_mainConfig, customDetails);
+                    }
+                    return sendCustomEvent(_evtName, _mainConfig, customDetails);
+
+                } catch (e) {
+                    // eslint-disable-next-line no-empty
+                }
+                return false;
+            }
+
+            function _updateEventListenerName(name?: string) {
+                try {
+                    _eventOff();
+                    if (name) {
+                        _evtName = name;
+                        _addEventListener();
+                    }
+                    return true;
+                } catch (e) {
+                    // eslint-disable-next-line no-empty
+                }
+                return false;
+            }
+
+
+            function _getFetchFnInterface() {
+                let _fetchFn = _overrideFetchFn;
+                if (isNullOrUndefined(_fetchFn)) {
+                    if (isFetchSupported()) {
+                        _fetchFn = _fetchSender;
+                    } else if (isXhrSupported()) {
+                        _fetchFn = _xhrSender;
+                    }
+                }
+                return _fetchFn;
+            }
+
+
+            function _fetchSender(url: string, oncomplete: OnCompleteCallback, isAutoSync?:  boolean) {
+                let global = getGlobal();
+                let fetchFn = (global && global.fetch) || null;
+                if (url && fetchFn && isFunction(fetchFn)) {
+                    try {
+                        const init: RequestInit = {
+                            method: STR_GET_METHOD
+                        };
+                        const request = new Request(url, init);
+                       
+                        fetch(request).then((response) => {
+                          
+                            if (response.ok) {
+                                response.text().then((text) => {
+                                    _doOnComplete(oncomplete, response.status, text, isAutoSync);
+                                })
+                            } else {
+                                _doOnComplete(oncomplete, response.status, null, isAutoSync);
+                            }
+
+                        }).catch((error) => {
+                            // eslint-disable-next-line no-empty
```

Comment:
This will still need to call the on complete otherwise the timer won't get re-started